src/main/java/io/mosip/mimoto/controller/IssuersController.java

```diff
@@ -78,4 +92,73 @@ public ResponseEntity<Object> getIssuerConfig(@PathVariable("issuer-id") String
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+        IssuerSupportedCredentialsResponse credentialsSupported;
+        try {
+            credentialsSupported = issuersService.getCredentialsSupported(issuerId, search);
+        }catch (ApiNotAccessibleException | IOException exception){
+            logger.error("Exception occurred while fetching issuers ", exception);
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }
+        responseWrapper.setResponse(credentialsSupported);
+
+        if (credentialsSupported.getSupportedCredentials().isEmpty()) {
+            logger.error("invalid issuer id passed - {}", issuerId);
+            responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_ISSUER_ID_EXCEPTION.getCode(), INVALID_ISSUER_ID_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(responseWrapper);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    }
+
+    @GetMapping("/{issuer-id}/credentials/{credentials-supported-id}/download")
+    public ResponseEntity<?> generatePdfForVCCredentials(@RequestHeader("Bearer") String token,
+                                                         @PathVariable("issuer-id") String issuerId,
+                                                         @PathVariable("credentials-supported-id") String credentialsSupportedId) {
+
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+
+        try{
+            IssuerDTO issuerConfig = issuersService.getIssuerConfig(issuerId);
+            CredentialIssuerWellKnownResponse credentialIssuerWellKnownResponse = issuersService.getCredentialWellKnownFromJson();
+            Optional<CredentialsSupportedResponse> credentialsSupportedResponse = credentialIssuerWellKnownResponse.getCredentials_supported().stream()
+                    .filter(credentialsSupported -> credentialsSupported.getId().equals(credentialsSupportedId))
+                    .findFirst();
+            if (credentialsSupportedResponse.isEmpty()){
+                logger.error("Invalid credential supported id passed - {}", credentialsSupportedId);
+                responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_CREDENTIAL_SUPPORTED_TYPE_ID_EXCEPTION.getCode(), INVALID_CREDENTIAL_SUPPORTED_TYPE_ID_EXCEPTION.getMessage())));
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
```

Comment:
passing invalid crendentialType should return **400 BadRequest**, **500** gives wrong impression.

Reply:
changed this